Products/GenericSetup/tool.py

```diff
@@ -1166,7 +1178,16 @@ def _runImportStepsFromContext(self,
 
         detect_steps = steps is None
 
-        for profile_id in chain:
+        last_index = len(chain) - 1
+        for index, profile_id in enumerate(chain):
+            if not always_apply_profiles and index != last_index:
```

Comment:
No, `always_apply_profiles` needs to be inside the loop. If it is False, then we do not need the check in the following lines, but can simple execute the original lines at the end of the loop (`context =...; self.applyContext(context)`).

I need the enumerate because I need the index.  Note that the last profile in the chain is the profile that you originally wanted to install (by selecting it in portal_setup / Import tab) and the earlier ones in the chain are its dependencies. For those dependencies we need the `always_apply_profiles check`, for the original one not, because the user has explicitly requested to apply this profile.